### PR TITLE
Move interned_strings and get build working

### PR DESCRIPTION
**Pull request by bwasti**
Summary: Refactoring out this diff D9819906

Differential Revision: D10024844

### CODE REVIEW

**Comment by ezyang** [comment]
Please ping when you want review

File: aten/src/ATen/core/interned_strings_class.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#include <vector>
+#include "ATen/core/Error.h"
+#include "ATen/core/interned_strings.h"
+#include "ATen/core/optional.h"
+#include "string.h"
+
+namespace torch {
+namespace jit {
+
+struct CAFFE2_API InternedStrings {
```
Comment (Yangqing) [inline aten/src/ATen/core/interned_strings_class.h]: Sorry, I am going to be a documentation policeman for a while but this class has zero comments :)

Comment (bwasti) [reply aten/src/ATen/core/interned_strings_class.h]: that's true https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/interned_strings_class.h

File: aten/src/ATen/core/interned_strings.h [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+#include <string>
+#include <unordered_map>
+#include <algorithm>
+
+#include <ATen/core/aten_interned_strings.h>
+#include <ATen/core/Macros.h>
+
+namespace torch { namespace jit {
+
+#if !AT_MOBILE
```
Comment (ezyang) [inline aten/src/ATen/core/interned_strings.h]: This deserves a comment, I think.

File: aten/src/ATen/core/interned_strings_class.h (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/core/interned_strings_class.h]:
```
// This class represents the global state that keeps track of interned strings.
// It maintains mappings of strings to their interned representation (Symbol)
// and vice versa.
```

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**